rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -670,6 +673,46 @@ public Topology getTopology(@HeaderParam("sessionid") String sessionId) throws N
         return rm.getSupportedNodeSourceInfrastructures();
     }
 
+    @Override
+    @GET
+    @GZIP
+    @Path("infrastructuresmapping")
+    @Produces("application/json")
+    public Map<String, List<String>> getInfrasToPoliciesMapping(@HeaderParam("sessionid") String sessionId)
+            throws NotConnectedException {
+        ResourceManager rm = checkAccess(sessionId);
+        Map<String, List<String>> result = rm.getInfrasToPoliciesMapping();
+        Collection<PluginDescriptor> supportedInfrastructures = rm.getSupportedNodeSourceInfrastructures();
+        Collection<PluginDescriptor> supportedPolicies = rm.getSupportedNodeSourcePolicies();
+        Iterator<Map.Entry<String, List<String>>> iterator = result.entrySet().iterator();
+        while (iterator.hasNext()) {
```

Comment:
Converting the result into a stream and using the filter() method can be a more concise method to consider here.

Reply:
yeah, i dont know why I did not do it. good point